mypy/stubtest.py

```diff
@@ -1044,8 +1036,13 @@ def verify_paramspecexpr(
         return
 
 
-def _verify_readonly_property(stub: nodes.Decorator, runtime: Any) -> Iterator[str]:
+def _verify_property(stub: nodes.Decorator, runtime: Any) -> Iterator[str]:
```

Comment:
This function was called `_verify_readonly_property` because it is only called on read-only properties, not because the purpose of the function is to verify whether a property is actually read-only or not.

Reply:
I think this needs a separate issue. Abstract settable properties are not fully supported: https://docs.python.org/3/library/abc.html#abc.abstractproperty

For example, this code is considered valid by mypy:

```python
import abc

class Base(abc.ABC):
    @property
    def x(self) -> int: ...

    @x.setter
    @abc.abstractmethod
    def x(self, arg: int) -> None: ...

class Child(Base):
    ...

c = Child()
c.x = 2
reveal_type(c)  # Revealed type is "ex.Child"
# Runtime:
# TypeError: Can't instantiate abstract class Child with abstract method x
```

I will open / search for related issues.

Reply:
Mypy correctly rejects this code, however:

```python
import abc

class Base(abc.ABC):
    @property
    def x(self) -> int: ...

    @x.setter
    @abc.abstractmethod
    def x(self, arg: int) -> None: ...

class Child(Base):
    ...

c = Child()  # error: Cannot instantiate abstract class "Child" with abstract attribute "x"
```

Reply:
Yeap, we can have three possible cases: abstract getter / abstract setter / both are abstract. I will report my findings soon in a new issue :)

Reply:
mypy does not know about deleters at all 😆

Reply:
https://github.com/python/mypy/issues/13649